### DISCUSSION TITLE
Failed import of old calculation results from remote computer

### DISCUSSION THREAD

**Discussion by ValkScripter** [aiidateam/aiida-core#6299] [Q&A]
Greetings developers,

I have dived into Aiida just a few days ago. 
I have learned and tested successfully the submission of CalcJobs on remote computers to run QuantumEspresso calculations.
So I wanted to make Aiida 'read' previous results that I had been running before. Here comes my problem.

If I understood correctly I should pass to the `CalcJob` (in my case through the quantumespresso.pw builder) the input named _remote_folder_ which is of the type `RemoteData` so that the job will not actually run the job but instead rewrite the inputs - that I should give again to the job - and retrieve the results from this _remote_folder_.
So I created this _remote_folder_ with the following command line:
```python
remote = RemoteData(remote_path='/absolute/path/to/folder/on/remote/computer', computer=load_computer("my_remote_computer"))
builder.remote_folder = remote
```
with the `builder` created from a code, itself belonging to a remote computer (with `core.ssh` transport and `core.pbspro` scheduler):
```python
builder = load_code("pw@my_remote_computer").get_builder()
```

I checked that the _remote_folder_ was working correctly with functional SSH connection in order for the `CalcJob` to read from it, by doing a :
```python
remote_folder.list_dir()
```
<details><summary>Results</summary>
<p>

```python
['file1.txt',
'prefix.pwi',
'prefix.pwo',
...
]
```

</p>
</details> 

After giving to the builder all the inputs and metadata that it needed (including _input_filename_ and _output_filename_),
```python
builder.metadata.options.input_filename = "prefix.pwi"
builder.metadata.options.output_filename = "prefix.pwo"
```

I launched the job (with either a **run** or a **submit**) and ultimately got an error with the following traceback:
```
+-> REPORT at ...
...
 |   File ".../aiida/lib/python3.12/site-packages/aiida/engine/processes/calcjobs/calcjob.py", line 636, in _perform_import
 |     retrieve_calculation(self.node, transport, retrieved_temporary_folder.abspath)
 |   File ".../aiida/lib/python3.12/site-packages/aiida/engine/daemon/execmanager.py", line 480, in retrieve_calculation
 |     transport.chdir(workdir)
 |   File ".../aiida/lib/python3.12/site-packages/aiida/transports/plugins/local.py", line 111, in chdir
 |     raise IOError(f"'{new_path}' is not a valid directory")
 | OSError: '/absolute/path/to/folder/on/remote/computer' is not a valid directory
```

I investigated and found that the __perform_import_ function in `aiida/engine/processes/calcjobs/calcjob.py` passes always - with the `with` statement - a `core.local` transport to the _retrieve_calculation_ function in file `aiida/engine/daemon/execmanager.py`, instead of a `core.ssh` transport, which I assume would be needed in my case.
I believe this is why the command `transport.chdir(workdir)` fails and that the given remote directory path is said to be not valid.
Below is the corresponding code part:

https://github.com/aiidateam/aiida-core/blob/cf091e80ff2b6aa03f41b56ba1976abb97298972/src/aiida/engine/processes/calcjobs/calcjob.py#L631-L639

1/ Am I right or am I missing something?
2/ In the former case, what should I do to circumvent this problem?
3/ In the latter case, what would be the missing piece?

Thank you for your time and attention.

**Comment by unkcpz** [comment]
Hi @ValkScripter, thanks for the question. The communication platform is moved to https://aiida.discourse.group/ see https://github.com/orgs/aiidateam/discussions/6113. 
Could you move the question there? Thanks!

**Reply by ValkScripter** [reply]
Hi @unkcpz, thank you for redirecting me to the communication platform. 
There is a question there giving me something that looks like an answer to my question already:
https://aiida.discourse.group/t/immigration-of-existing-vasp-calculations/326/2
Thanks again!
